internal/discovery/saptune.go

```diff
@@ -0,0 +1,64 @@
+package discovery
+
+import (
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/trento-project/agent/internal/core/saptune"
+	"github.com/trento-project/agent/internal/discovery/collector"
+	"github.com/trento-project/agent/pkg/utils"
+)
+
+const SaptuneDiscoveryID string = "saptune_discovery"
+const SaptuneDiscoveryMinPeriod time.Duration = 1 * time.Second
+
+type SaptuneDiscovery struct {
+	id              string
+	collectorClient collector.Client
+	interval        time.Duration
+}
+
+type SaptuneDiscoveryPayload struct {
+	PackageVersion   string `json:"package_version"`
+	SaptuneInstalled bool   `json:"saptune_installed"`
+	Status           string `json:"status"`
+}
+
+func NewSaptuneDiscovery(collectorClient collector.Client, config DiscoveriesConfig) Discovery {
+	return SaptuneDiscovery{
+		id:              SaptuneDiscoveryID,
+		collectorClient: collectorClient,
+		interval:        config.DiscoveriesPeriodsConfig.Saptune,
+	}
+}
+
+func (d SaptuneDiscovery) GetID() string {
+	return d.id
+}
+
+func (d SaptuneDiscovery) GetInterval() time.Duration {
+	return d.interval
+}
+
+func (d SaptuneDiscovery) Discover() (string, error) {
+	saptuneRetriever, err := saptune.NewSaptune(utils.Executor{})
+	if err != nil {
+		return "", err
+	}
+
+	saptuneData, _ := saptuneRetriever.RunCommand("--format", "json", "status")
```

Comment:
We cannot ignore the error.
What if json format is not supported?
That's why I like to have the `IsJsonSupported` field.
Because you can check it here, and if it is false, send a payload with `status: nil` without even using `RunCommand`

Reply:
Alright, I understood now.
You are redirecting the nil coming from the `RunCommand`, a bit ugly for me, but it should work